src/pveclib/vec_f128_ppc.h

```diff
@@ -951,17 +1314,352 @@ test_cmpltf128_v3d (vui128_t vfa128, vui128_t vfb128)
+static inline vec_xscvuqqp (vui128_t int128)
+{
+  __binary128 result;
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
+  __asm__(
+      "xscvuqqp %0,%1"
+      : "=v" (result)
+      : "v" (int128)
+      : );
+#elif defined (_ARCH_PWR9) && defined (__FLOAT128__) && (__GNUC__ > 7)
+  vui64_t int64 = (vui64_t) int128;
+  __binary128 hi64, lo64;
+  __binary128 two64 = 0x1.0p64;
+  hi64 = int64[VEC_DW_H];
+  lo64 = int64[VEC_DW_L];
+  result = (hi64 * two64) + lo64;
+#elif  defined (_ARCH_PWR8)
+...
+#endif
+  return result;
+}
+ * \endcode
+ * The POWER10 implementation uses the <B>xscvuqqp</B> instruction.
+ * While POWER9 implementation uses <B>xscvudqp</B> instructions to
+ * convert the high/low 64-bit halves of the quadword integer.
+ * To complete the conversion we need to multiply the converted high
+ * 64-bits by 2**64 than add the lower converted 64-bits.
+ * The compiler should generate something like this:
+ * \code
+<test_vec_xscvuqqp_PWR9>:
+     addis   r9,r2,0 ## R_PPC64_TOC16_HA   .rodata."0x1.0p64"
+     addi    r9,r9,0 ## R_PPC64_TOC16_LO   .rodata."0x1.0p64"
+     xxspltd v0,v2,1
+     xscvudqp v2,v2
+     xscvudqp v0,v0
+     lxv     v1,0(r9)
+     xsmaddqp v2,v0,v1
+     blr
+ * \endcode
+ *
+ * The POWER8 implementation looks like this:
+ * \code
+#elif  defined (_ARCH_PWR8)
+  vui64_t q_exp;
+  vui128_t q_sig;
+  const vui128_t q_zero = (vui128_t) { 0 };
+  const vui32_t lowmask = CONST_VINT128_W( 0, 0, 0, 1);
+
+  q_sig = int128;
+  // Quick test for 0UL as this case requires a special exponent.
+  if (vec_cmpuq_all_eq (q_sig, q_zero))
+    {
+      result = vec_xfer_vui128t_2_bin128 (q_zero);
+    }
+  else
+    { // We need to produce a normal QP, so we treat the QW integer
+      // like a denormal, then normalize it.
+      // Start with the quad exponent bias + 127 then subtract the count of
+      // leading '0's. The 128-bit sig can have 0-127 leading '0's.
+      vui64_t q_expm = (vui64_t) CONST_VINT64_DW(0, (0x3fff + 127));
+      vui64_t i64_clz = (vui64_t) vec_clzq (q_sig);
+      q_sig = vec_slq (q_sig, (vui128_t) i64_clz);
+      q_exp = vec_subudm (q_expm, i64_clz);
+      // This is the part that might require rounding.
+      // For example Round to zero
+      // Shift right 15-bits to normalize and truncate
+      q_sig = vec_srqi ((vui128_t) q_sig, 15);
+      //...
+      q_exp = vec_swapd (q_exp);
+      result = vec_xsiexpqp (q_sig, q_exp);
+    }
+ * \endcode
+ * In this example the Significand (including the L-bit) is right
+ * justified in the high-order 113-bits of q_sig.
+ * The guard, round, and sticky (GRX) bits are in the low-order
+ * 15 bits.
+ * The sticky-bits are the last 13 bits and are logically ORed
+ * (or added to 0x1fff) to produce the X-bit.
+ *
+ * The signed quadword conversion is bit more complicated for both
+ * POWER9/8. For example:
+ * \code
+__binary128
+static inline vec_xscvsqqp (vi128_t int128)
+{
+  __binary128 result;
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
```

Comment:
`__GNUC__` again. This is a global question, so I won't add more comments.

Reply:
Yes it is needed